internal/cli/commands.go

```diff
@@ -103,3 +108,30 @@ func (commands Commands) Sort() Commands {
 
 	return commands
 }
+
+// SetCategory sets the given `category` for the `commands`.
+func (commands Commands) SetCategory(category *Category) Commands {
```

Comment:
Noticed that struct Commands has methods on both value and pointer receivers - it is by intention?

Reply:
Yes, I do not want to use pointers for the slice, but in order to append a new command, I have to

```go
func (commands *Commands) Add(cmd *Command) {
    *commands = append(*commands, cmd)
}
```